### PR TITLE
Add callback to add dynamic properties after navigation

### LINKED ISSUE #110: Dynamically setting additional properties on route change
## Request / Idea

Allow `meta.gtmAdditionalEventData` to be dynamically resolved whenever `router.afterEach` is called. This could be done for example by allowing `gtmAdditionalEventData` to be a function or by adding a callback to the options that is called to resolve `gtmAdditionalEventData`.

## Additional Context

Currently additional properties  can be set on a per route basis with `meta.gtmAdditionalEventData`. I would like to read these properties from the vuex store when the route changes because the data changes over time.

### PR DESCRIPTION
**Pull request by lukahartwig**
Closes #110 

This PR adds the ability to define a callback to dynamically compute custom properties to add to the page view event after navigation.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -119,7 +131,10 @@ async function initVueRouterGuard(
       }
     }
 
-    const additionalEventData: Record<string, any> = (to.meta?.gtmAdditionalEventData as Record<string, any>) ?? {};
+    const additionalEventData: Record<string, any> = {
+      ...deriveAdditionalEventData(to, from),
```
Comment (Shinigami92) [inline src/index.ts:135]: We are already in an async context here. I think we can improve this with letting `deriveAdditionalEventData` be an async function and await it here.

We need to typedef `vueRouterAdditionalEventData` as `?: (to: RouteLocationNormalized, from: RouteLocationNormalized) => Record<string, any> | Promise<(to: RouteLocationNormalized, from: RouteLocationNormalized) => Record<string, any>>;` for this. Or maybe we should extract the callback type to a type and then reuse it.

Hope you know what I mean :D

**Review by Shinigami92 — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (lukahartwig) [reply src/index.ts:135]: This happens within the callback to `vueRouter.afterEach` which is currently not `async`. Only the outer function is at the moment. We can make the callback to `vueRouter.afterEach` asynchronous though, I think that should work. What do you think?

Comment (Shinigami92) [reply src/index.ts:135]: I think thios would be a nice to have / cool addition :slightly_smiling_face:

Comment (lukahartwig) [reply src/index.ts]: Added the suggestion

**Review by Shinigami92 — APPROVED** [review]
Would you like to add an additional test for the new async strategy?

File: src/index.ts (hunk 1 above)
Comment (Shinigami92) [reply src/index.ts]: nice :raised_hands:

**Comment by lukahartwig** [comment]
Done

**Comment by Shinigami92** [comment]
Will merge it, but cause it's so late here in our town (du bist ja auch in Hamburg ^^) I will release it tomorrow.

**Comment by lukahartwig** [comment]
No problem, and thanks for responding so fast.

Nordische Grüße 👋

**Pull request merged**